### ISSUE TITLE
Tags widgets

### ISSUE THREAD

**Issue by richardolsson** [zetkin/app.zetkin.org#607]
We need a reusable widget (or set of widgets) to create, edit, assign and "unassign" tags to objects (e.g. people or journeys).

## Definitions of key terminology
* An **"object"** in this context means anything that can be tagged, e.g. a person or a journey (in the future, many more)
* A **tag** is a data model / API resource that can be linked/assigned to objects. Normal tags are binary, i.e. can be either assigned or not assigned to an object.
* A **value tag** is a special type of tag that besides being assigned to an object, can also hold a value specific to that object, i.e. the same tag can hold different values when assigned to different objects

## Target audience
These are the roles that will most likely use this feature:

* Basically anyone who use Zetkin for organizing, case work etc will use tags from time to time.

## Requirements (job stories)
No job stories. This widget will be used in multiple different places (where job stories apply) such as #556 and #578.

## Solution
A complete UX has been designed in Figma:
https://www.figma.com/file/YFXcYN3LUhHghLb0cFqU85/Casework?node-id=1571%3A64584

![image](https://user-images.githubusercontent.com/550212/158355795-d50f620a-45b5-4406-87e7-a220df5acadb.png)

## Tasks
- [x] #608
- [x] #609
- [x] #610
- [x] #611
- [x] #612
- [x] #632

**Comment by richardolsson** [comment]
Boom! Epic closing. 💥
